### PR TITLE
[WFCORE-4228] Add methods to the management API to get Capability Runtime API as an Optional object

### PR DESCRIPTION
**Pull request by yersan**
- Required changes in wildfly core for WFLY-11186
- Add additional method signature to allow get an optional capability API

Jira issue: https://issues.jboss.org/browse/WFCORE-4228

### CODE REVIEW

**Comment by yersan** [comment]
Retest this please

**Comment by bstansberry** [comment]
If @jmesnil is ok with this API addition, can you update the commit message, PR title, PR and JIRA description to match the current scope of the change?

File: controller/src/main/java/org/jboss/as/controller/capability/CapabilityServiceSupport.java [hunk 1]
```diff
@@ -86,6 +88,21 @@ public NoSuchCapabilityException(String message) {
      */
     <T> T getCapabilityRuntimeAPI(String capabilityBaseName, String dynamicPart, Class<T> apiType) throws NoSuchCapabilityException;
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/capability/CapabilityServiceSupport.java]: For consistency, this variant should have an Optional version too.

Comment (yersan) [reply controller/src/main/java/org/jboss/as/controller/capability/CapabilityServiceSupport.java]: I added the missing variant using optional, it is still pending for a review from Jeff.

**Pull request merged**